cpp/tensorrt_llm/executor/cache_transmission/cacheSplitConcat.cu

```diff
@@ -1565,7 +1592,14 @@ void concatKVCache(std::vector<runtime::ITensor::SharedPtr> const& inputSplitBlo
     int const numLayers = selfParallelConfig.mAttentionLayerNumPerPP.at(selfPPRank);
     TLLM_LOG_DEBUG(mpi::MpiComm::world().getRank(), "concatKVCache numLayers:%d", numLayers);
     int const headNum = selfModelConfig.mNbKvHeadsPerLayer[0];
-    int const dimsPerHead = selfModelConfig.mSizePerHead;
+    int dimsPerHead = selfModelConfig.mSizePerHead;
+    if (isIndexerKCache)
+    {
+        dimsPerHead = selfCacheState.getIndexerDimPerHead()
+            + selfCacheState.getIndexerDimPerHead() / selfCacheState.getIndexerKCacheQuantBlockSize() * 4;
+        sizePerHead = dimsPerHead;
+    }
+
```

Comment:
This is same duplicated code as above. Can we refactor to avoid duplication?

Reply:
refactored.